src/Services/AbstractResourceService.php

```diff
@@ -172,15 +169,25 @@ public function search(array $payload, array $options = [])
      *
      * @param string $method
      * @param string $url
-     * @param array $options
+     * @param ResourceServiceOptions $options
      *
-     * @return \LiveIntent\Resource
+     * @return \LiveIntent\Resource|ResourceResponse|null
      */
-    public function request(string $method, string $url, array $options = [])
+    public function requestWithOptions(string $method, string $url, ResourceServiceOptions $options = null)
     {
         $response = $this->requestRaw($method, $url, $options);
+        $json = $response->json();
+
+        $resource = null;
+        if (array_key_exists('output', $json)) {
+            $resource = $this->newResource($json['output']);
+        }
```

Comment:
Actually, we can't make this change as it will break backwards compatibility by allowing this to return `null` now.

Reply:
it should not break things, as `$resource` can be null only when `keepRawResponse` is true